pkg/clustermanager/manager.go

```diff
@@ -319,14 +316,10 @@ func ToRESTConfig(cluster *v3.Cluster, context *config.ScaledContext) (*rest.Con
 			}
 			if cluster.Status.Driver == "googleKubernetesEngine" && cluster.Spec.GenericEngineConfig != nil {
 				cred, _ := (*cluster.Spec.GenericEngineConfig)["credential"].(string)
-				ts, err := gke.GetTokenSource(context.RunContext, cred)
-				if err == nil {
-					return &oauth2.Transport{
-						Source: ts,
-						Base:   rt,
-					}
+				rt, err = gke.Oauth2Transport(context.RunContext, rt, cred)
+				if err != nil {
+					logrus.Errorf("unable to retrieve token source for GKE oauth2: %v", err)
 				}
-				logrus.Errorf("unable to retrieve token source for GKE oauth2: %v", err)
```

Comment:
So no changes here, just consolidating.

Reply:
Correct. Since we are doing this in two different places I created a function to be called.